probe-rs/src/core.rs

```diff
@@ -163,6 +165,35 @@ pub trait CoreInterface: MemoryInterface + ExceptionInterface {
     }
 }
 
+/// A snapshot representation of a core state.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct CoreDump {
+    stack_data: Vec<u8>,
+    heap_data: Vec<u8>,
```

Comment:
How do you know at which address the heap is?

Maybe all memory could just be stored like the additional memory?

Reply:
The idea was that one passes it to the function. Maybe we can automate it in the future, but my goal was to not read the entire chip RAM when it's not required.
The additional memory was meant for reading specific peripheral registers later on so we do not have to scan `2^32` addresses :P

Good idea on storing all memory the same, will do :)

Reply:
I wonder if we could somehow extract those things from the ELF.